test/src/test/java/hudson/model/labels/LabelAtomTest.java

```diff
@@ -20,5 +26,57 @@ public void selfLabel() throws Exception {
         assertThat(self.isSelfLabel(), is(true));
         Label label = new LabelAtom("label");
         assertThat(label.isSelfLabel(), is(false));
+        Label selfJenkins = j.jenkins.getLabel("node");
+        assertThat(selfJenkins.isSelfLabel(), is(true));
+    }
+
+    @Test
+    public void getNodes() throws Exception {
+        Node n1 = j.createSlave("n1", "label", null);
+        Node n2 = j.createSlave("n2", "label label2", null);
+        Node n3 = j.createSlave("n3", "label2", null);
+        Label l = j.jenkins.getLabel("label");
+        Label l2 = j.jenkins.getLabel("label2");
+        Label l3 = j.jenkins.getLabel("label3");
+        assertThat(l.getNodes().size(), is(2));
+        assertThat(l.getNodes(), containsInAnyOrder(n1,n2));
+        assertThat(l2.getNodes().size(), is(2));
+        assertThat(l2.getNodes(), containsInAnyOrder(n3,n2));
+        assertThat(l3.getNodes().isEmpty(), is(true));
```

Comment:
Failures may be clearer with
```suggestion
        assertThat(l3.getNodes() is(empty()));
```